src/darwin/Framework/CHIPTests/MTRFabricInfoTests.m

```diff
@@ -0,0 +1,283 @@
+        @"nodeID" : nodeID,
+        @"label" : @"",
+        @"hasIntermediateCertificate" : @(NO),
+        @"fabricIndex" : @(1)
+    } ]));
+
+    [factory stopControllerFactory];
+    XCTAssertFalse([factory isRunning]);
+}
+
+- (void)testFabricInfoTwoFabrics
+{
+    __auto_type * factory = [MTRDeviceControllerFactory sharedInstance];
+    XCTAssertNotNil(factory);
+    XCTAssertFalse([factory isRunning]);
+
+    __auto_type * storage = [[MTRTestStorage alloc] init];
+    __auto_type * factoryParams = [[MTRDeviceControllerFactoryParams alloc] initWithStorage:storage];
+    XCTAssertTrue([factory startControllerFactory:factoryParams error:nil]);
+    XCTAssertTrue([factory isRunning]);
+
+    __auto_type * testKeys1 = [[MTRTestKeys alloc] init];
+    XCTAssertNotNil(testKeys1);
+
+    const unsigned fabricID1 = 8;
+    __auto_type * params = [[MTRDeviceControllerStartupParams alloc] initWithIPK:testKeys1.ipk
+                                                                        fabricID:@(fabricID1)
+                                                                       nocSigner:testKeys1];
+    XCTAssertNotNil(params);
+
+    params.vendorID = @(kTestVendorId1);
+
+    MTRDeviceController * controller1 = [factory createControllerOnNewFabric:params error:nil];
+    XCTAssertNotNil(controller1);
+    XCTAssertTrue([controller1 isRunning]);
+
+    __auto_type * rootKeys = [[MTRTestKeys alloc] init];
+    XCTAssertNotNil(rootKeys);
+
+    __auto_type * root = [MTRCertificates createRootCertificate:rootKeys issuerID:nil fabricID:nil error:nil];
+    XCTAssertNotNil(root);
+
+    __auto_type * intermediateKeys = [[MTRTestKeys alloc] init];
+    XCTAssertNotNil(intermediateKeys);
+
+    __auto_type * intermediate = [MTRCertificates createIntermediateCertificate:rootKeys
+                                                                rootCertificate:root
+                                                          intermediatePublicKey:intermediateKeys.publicKey
+                                                                       issuerID:nil
+                                                                       fabricID:nil
+                                                                          error:nil];
+    XCTAssertNotNil(intermediate);
+
+    const unsigned fabricID2 = 17;
+    params = [[MTRDeviceControllerStartupParams alloc] initWithIPK:rootKeys.ipk fabricID:@(fabricID2) nocSigner:intermediateKeys];
+    XCTAssertNotNil(params);
+
+    params.vendorID = @(kTestVendorId2);
+    params.rootCertificate = root;
+    params.intermediateCertificate = intermediate;
+
+    MTRDeviceController * controller2 = [factory createControllerOnNewFabric:params error:nil];
+    XCTAssertNotNil(controller2);
+    XCTAssertTrue([controller2 isRunning]);
+
+    __auto_type * nodeID1 = [controller1 controllerNodeID];
+    __auto_type * nodeID2 = [controller2 controllerNodeID];
+
+    __auto_type * fabricInfoList = [factory knownFabrics];
```

Comment:
Is it intentional that the test is expecting the fabrics in a fixed order? The API doesn't say anything about any particular ordering.

Reply:
Otherwise just note it in the test, and we can fix it if it becomes an issue.